drivers/adc/adc_stm32.c

```diff
@@ -592,7 +592,14 @@ static int start_read(const struct device *dev,
 	!defined(CONFIG_SOC_SERIES_STM32F1X) && \
 	!defined(STM32F3X_ADC_V2_5) && \
 	!defined(CONFIG_SOC_SERIES_STM32L1X)
+
+		/* we cannot calibrate the ADC while the ADC is enabled */
+		LL_ADC_Disable(adc);
+		while (LL_ADC_IsEnabled(adc) == 1UL) {
+		}
 		adc_stm32_calib(dev);
+		/* re-enable ADC after calibration */
+		LL_ADC_Enable(adc);
```

Comment:
These changes are part of another PR https://github.com/zephyrproject-rtos/zephyr/pull/40937 that just needs to be merged